app/src/main/java/com/kickstarter/viewmodels/ProjectCardHolderViewModel.java

```diff
@@ -0,0 +1,279 @@
+
+    public ViewModel(final @NonNull Environment environment) {
+      super(environment);
+
+      final Observable<Boolean> projectInEndedStates = this.project
+        .map(
+          p -> p.state().equals(Project.STATE_SUCCESSFUL)
+            || p.state().equals(Project.STATE_CANCELED)
+            || p.state().equals(Project.STATE_FAILED)
+            || p.state().equals(Project.STATE_SUSPENDED)
+        );
+
+      this.projectOutput = this.project;
+
+      this.backersCountText = this.project
+        .map(Project::backersCount)
+        .map(NumberUtils::format);
+
+      this.backingViewGroupIsGone = this.project
+        .map(p -> p.isBacking() || p.isStarred() || p.isPotdToday() || p.isFeaturedToday());
+
+      this.blurbText = this.project
+        .map(Project::blurb);
+
+      this.categoryNameText = this.project
+        .map(Project::category)
+        .map(Category::name)
+        .compose(coalesce(""));
+
+      this.deadlineCountdownText = this.project
+        .map(ProjectUtils::deadlineCountdownValue)
+        .map(NumberUtils::format);
+
+      this.featuredViewGroupIsGone = this.project
+        .map(p -> p.isBacking() || p.isStarred() || p.isPotdToday() || !p.isFeaturedToday());
+
+      this.friendAvatarUrl = this.project
+        .filter(Project::isFriendBacking)
+        .map(Project::friends)
+        .map(friends -> friends.get(0).avatar().small());
+
+      this.friendBackingViewIsHidden = this.project
+        .map(Project::isFriendBacking)
+        .map(BooleanUtils::negate);
+
+      this.friendsForNamepile = this.project
+        .filter(Project::isFriendBacking)
+        .map(Project::friends);
+
+      this.fundingUnsuccessfulTextViewIsGone = this.project
+        .map(p ->
+          !p.state().equals(Project.STATE_CANCELED) &&
+            !p.state().equals(Project.STATE_FAILED) &&
+            !p.state().equals(Project.STATE_SUSPENDED));
+
+      this.imageIsInvisible = this.project
+        .map(Project::photo)
+        .map(ObjectUtils::isNull);
+
+      this.metadataViewGroupIsGone = this.project
+        .map(p -> !p.isBacking() && !p.isStarred() && !p.isPotdToday() && !p.isFeaturedToday());
+
+      this.nameText = this.project
+        .map(Project::name);
+
+      this.notifyDelegateOfProjectClick = this.project
+        .compose(Transformers.takeWhen(this.projectClicked));
+
+      this.percentageFunded = this.project
+        .map(Project::percentageFunded)
+        .map(ProgressBarUtils::progress);
+
+      this.percentageFundedProgressBarIsGone = projectInEndedStates;
+
+      this.percentageFundedText = this.project
+        .map(Project::percentageFunded)
+        .map(NumberUtils::flooredPercentage);
+
+      this.photoUrl = this.project
+        .map(Project::photo)
+        .filter(ObjectUtils::isNotNull)
+        .map(Photo::med);
+
+      this.potdViewGroupIsGone = this.project
+        .map(p -> p.isBacking() || p.isStarred() || !p.isPotdToday() || p.isFeaturedToday());
```

Comment:
after having seen this `backing || starred || potd || featured` logic 4 times I'm starting to think there's a better way! what if we created a lil enum to express the four types of metadata there could be, and then we can localize the cascading logic into a single place, and derive the outputs from that?

hilariously, that is precisely how @luoser did it 9 months ago in swift! https://github.com/kickstarter/ios-oss/blob/master/Library/ViewModels/DiscoveryPostcardViewModel.swift